src/angulartics.js

```diff
@@ -223,8 +227,51 @@ function $analyticsRun($rootScope, $window, $analytics, $injector) {
     return false;
   }
 
+  function arrayDifference(a1, a2) {
+    var result = [];
+    for (var i = 0; i < a1.length; i++) {
+      if (a2.indexOf(a1[i]) === -1) {
+        result.push(a1[i]);
+      }
+    }
+    return result;
+  }
+
+  function filterQueryString(url,keysMatchArr,containerEvalStr){
+    if (/\?/.test(url) && keysMatchArr.length > 0) {
+      var urlArr = url.split('?');
+      var urlBase = urlArr[0];
+      var pairs = urlArr[1].split('&');
+      var matchedPairs = [];
+
+      for (var i = 0; i < keysMatchArr.length; i++) {
+        var listedKey = keysMatchArr[i];
+        for (var j = 0; j < pairs.length; j++) {
+          if ((listedKey instanceof RegExp && listedKey.test(pairs[j])) || pairs[j].indexOf(listedKey) > -1) matchedPairs.push(pairs[j]);
+        }
+      }
+      if (eval(containerEvalStr).length > 0) {
```

Comment:
I'm ok with it's use as an internal - just don't want to promote eval use in general. Unless any @angulartics/core contrib has a counter point or an alternative I'll merge this later this week